nym-vpn-lib/src/platform/android.rs

```diff
@@ -300,6 +393,29 @@ fn try_sending_random_udp(is_ipv6_enabled: bool) -> Result<(), SendRandomDataErr
     Ok(())
 }
 
+fn parse_api_url_from_java(env: &JnixEnv, api_url: JString<'_>) -> Url {
+    match Url::from_str(&String::from_java(env, api_url)) {
+        Err(err) => {
+            error!("Could not parse api_url: {:?}. Using the default.", err);
+            gateway_client::Config::default().api_url
+        }
+        Ok(url) => url,
+    }
+}
+
+fn parse_explorer_url_from_java(env: &JnixEnv, explorer_url: JString<'_>) -> Option<Url> {
+    match Url::from_str(&String::from_java(env, explorer_url)) {
+        Err(err) => {
+            error!(
+                "Could not parse explorer_url: {:?}. Using the default.",
+                err
+            );
+            gateway_client::Config::default().explorer_url
```

Comment:
See comment above, imho we should probably return an error here instead of using the (likely wrong) default value